bridge-client/src/iosMain/kotlin/org/sagebionetworks/bridge/kmm/shared/NativeTimelineManager.kt

```diff
@@ -64,6 +91,16 @@ abstract class AbstractNativeTimelineManager(
         }
     }
 
+    fun getAllCachedAdherence(callBack: (List<NativeAdherenceRecord>) -> Unit) {
```

Comment:
We avoid "get" in the iOS code because that word already has a different meaning. Below when getting the AssessmentConfig we use "fetch" to indicate that there is a callback.

That said, I'm not sure why you are adding this here. Not certain what your UI/UX is that requires it, but this is the purpose of the `NativeHistoryManager`.

```suggestion
    fun fetchAllCachedAdherence(callBack: (List<NativeAdherenceRecord>) -> Unit) {
```

Reply:
Switched over to the observe pattern.